config/runtime.exs

```diff
@@ -3,9 +3,30 @@ import Config
 config :alert_processor, AlertProcessor.Repo,
   url: System.fetch_env!("DATABASE_URL_#{config_env() |> to_string() |> String.upcase()}")
 
+config :concierge_site, ConciergeSite.Endpoint,
+  authentication_source: System.get_env("AUTHENTICATION_SOURCE", "local")
+
+config :concierge_site,
+  keycloak_base_uri: System.get_env("KEYCLOAK_BASE_URI"),
+  keycloak_client_id: System.get_env("KEYCLOAK_CLIENT_ID"),
+  keycloak_redirect_uri: System.get_env("KEYCLOAK_REDIRECT_URI")
+
+config :ueberauth, Ueberauth.Strategy.OIDC,
+  keycloak: [
+    fetch_userinfo: true,
+    userinfo_uid_field: "preferred_username",
+    discovery_document_uri: System.get_env("KEYCLOAK_WELL_KNOWN_OIDC"),
+    client_id: System.get_env("KEYCLOAK_CLIENT_ID"),
+    client_secret: System.get_env("KEYCLOAK_CLIENT_SECRET"),
+    redirect_uri: System.get_env("KEYCLOAK_REDIRECT_URI"),
+    logout_uri: System.get_env("KEYCLOAK_LOGOUT_URI"),
```

Comment:
note(non-blocking): @lemald and I were just talking about implementing this on mycharlie and that maybe this(`config :ueberauth, Ueberauth.Strategy.OIDC`) isn't the best place for this config item(`logout_uri`) since Ueberauth doesn't handle the logout flow? (@lemald knows more about this than I)

If you're looking for ideas on where else to put it, I think I'm going to play with the idea of adding `logout_uri` config to `MyCharlieWeb.AuthController` on our side.

Reply:
Yeah, that's a fair point. It's kind of funky because it relates to Keycloak, but as you say, not actually to Ueberauth.

Reply:
Another note: I think I only learned about this in-depth after implementing this logout URI var, but Keycloak publishes a list of ["well-known" endpoints](https://login-sandbox.mbtace.com/auth/realms/MBTA/.well-known/openid-configuration). (It seems like that's probably part of the OIDC standard, since we pass that down to the [openid_connect](https://github.com/DockYard/openid_connect) library.) And that list does include that logout URL under `end_session_endpoint`.  It would be more effort, so maybe not worth it, but maybe the more "proper" thing would be to query that well-known list and discover the logout URL instead of saving it. (But all that said, I'm probably not going to bother with that right this minute…)

Reply:
Oh interesting, This'll be helpful for me exploring logout on our side, thanks!!

Reply:
Poking around a little, maybe you could even [get it from openid_connect](https://github.com/DockYard/openid_connect/blob/ddafedc3c81a5bc91919d13630b6cb5ea2595ffe/lib/openid_connect.ex#L152)? That library is wrapped by [ueberauth_oidc](https://github.com/DefactoSoftware/ueberauth_oidc), but I did end up needing to use it directly in [one place](https://github.com/mbta/alerts_concierge/pull/1309/files#diff-bb7878acb7873ea7a1866a588482aee73b3b4ee08c2fbeb3e4e6b496b88d3c5dR121).

Reply:
Heh, looks like the library was rewritten like a month ago, and that function seems to be new? https://github.com/DockYard/openid_connect/pull/50

Reply:
Oh, interesting. And while I was working I was constantly reference the source files loaded locally, so that makes sense, I probably looked around it it wasn't there.